### PR TITLE
[mlflow/R] Update R client to call record logged model 

### PR DESCRIPTION
**Pull request by tomasatdatabricks**
## What changes are proposed in this pull request?
Followup to #2369
Update R client to call new api to record logged model metadata with the tracking server.

## How is this patch tested?
Unit test.

## Release Notes
n/a
### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Followup to previous PR, adds functionality to R client.
### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [x] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/R/mlflow/R/model.R [hunk 1]
```diff
@@ -28,8 +29,22 @@ mlflow_save_model <- function(model, path, ...) {
...
-  mlflow_save_model(model, path = temp_path, ...)
-  mlflow_log_artifact(path = temp_path, artifact_path = artifact_path)
+  model_spec <- mlflow_save_model(model, path = temp_path, model_spec = list(
+    utc_time_created = mlflow_timestamp(),
+    run_id = mlflow_get_active_run_id(),
+    artifact_path = artifact_path,
+    flavors = list()
+  ), ...)
+  res <- mlflow_log_artifact(path = temp_path, artifact_path = artifact_path)
+  mlflow_record_logged_model(model_spec)
```
Comment (smurching) [inline mlflow/R/mlflow/R/model.R]: Should we try-catch this to handle the case where the server doesn't have the log-model endpoint implemented?

**Review by smurching — COMMENTED** [review]
Just one comment around adding a try-catch, looks good otherwise!

File: mlflow/R/mlflow/R/model.R (hunk 1 above)
Comment (tomasatdatabricks) [reply mlflow/R/mlflow/R/model.R]: yes, we should! good catch

File: mlflow/server/handlers.py [hunk 2]
```diff
@@ -453,7 +453,6 @@ def _log_batch():
 def _log_model():
     request_message = _get_request_message(LogModel())
     try:
-        _logger.error(request_message.model_json)
```
Comment (tomasatdatabricks) [inline mlflow/server/handlers.py]: This was forgotten in earlier python PR, we should remove it.

**Review by smurching — APPROVED** [review]
LGTM

**Pull request merged**